graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -78,24 +75,16 @@ def _validate_macro_ast_directives(ast, inside_fold_scope=False):
         inside_fold_scope: bool, whether the subtree is within a @fold scope
     """
     subselection_inside_fold_scope = inside_fold_scope
-    directives_with_no_restrictions = frozenset({
-        FilterDirective.name,
-        TagDirective.name,
-        OptionalDirective.name,
-        RecurseDirective.name,
-        MacroEdgeDefinitionDirective.name,
-    })
-    disallowed_directives = frozenset({
-        OutputDirective.name,
-        OutputSourceDirective.name,
-    })
+    optional_directive_names = [directive.name
```

Comment:
Two things here:
- When a comprehension of any kind (list, set, dict, ...) requires multiple lines, we always prefer the following formatting:
```
foo = [
    bar
    for bar in baz
    if quux
]
```
- On line 84 below, you use the `in` operator against this variable. Why not make it a set instead of using a list? Let's not pay `O(n)` runtime costs if we don't have to.

Reply:
One more -- reconsider the name `optional_directive_names`. Ignoring the code that computes its value, which of these do you think the reader will think it refers to:
- names of the `@optional` directive
- names of directives which are somehow optional
- names, if they happen to exist, of directives

Reply:
- Since there's now an allowed directive list, I've renamed this variable to names_of_allowed_directives.